dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -211,3 +258,83 @@ func validateFromChainState(runtimeInstance parachainruntime.RuntimeInstance, po
 
 	return &candidateCommitments, persistedValidationData, isValid, nil
 }
+
+// validateFromExhaustive validates a candidate parachain block with provided parameters
+func validateFromExhaustive(validationHost parachainruntime.ValidationHost,
+	persistedValidationData parachaintypes.PersistedValidationData,
+	validationCode parachaintypes.ValidationCode,
+	candidateReceipt parachaintypes.CandidateReceipt, pov parachaintypes.PoV) (
+	*ValidationResult, error) {
+
+	validationCodeHash := validationCode.Hash()
+	// basic checks
+	err := performBasicChecks(&candidateReceipt.Descriptor, persistedValidationData.MaxPovSize, pov,
+		validationCodeHash)
+	if err != nil {
+		return &ValidationResult{ //nolint:nilerr  // note: we don't want to return an error here as the error is already set
+			IsValid: false,
+			Err:     err,
+		}, nil
```

Comment:
Can't set this `err` as a reason for invalidity, 

`performBasicChecks` not only returns the reason for invalidity, but it also returns internal errors like hashing errors and encoding errors.

Reply:
Yeah,  I do not see reason of returning whole structure when err could be enought to understand that structure is not valid, if you need exactly undrstand this, there could be 3rd bool param in return params

Reply:
@P1sar `ValidationResult` is the type we send to other subsystems. So, we need to return this struct and an IsValid field for more readability. 
- I was confused about the error returned from `performBasicChecks`.
---
@edwardmack, we can return multiple errors from `performBasicChecks`. one for a reason for invalidity and the other for internal errors. (add a comment explaining return types).

Reply:
I don't see the need to keep an error in both `ValidationResult` and `parachaintypes.OverseerFuncRes`.  What is the difference to the receiver of this `OverseerFuncRes` type over a channel whether the error is within the `ValidationResult` or the `OverseerFuncRes` error.  I would assume you can omit the error `ValidationResult` type completely and just send back an `OverseerFuncRes[bool]` type to represent validity (IsValid) and an error.  You can leave it up to the function caller/consumer of the message to determine next steps based on the error returned using `errors.Is/As`

Reply:
@axaysagathiya I'll update `performBasciChecks` to return multiple errors.  What are your thought regarding removing err from `ValidationResult` (See comment below)?

Reply:
@edwardmack @timwu20 
- An error field in the `ValidationResult` is the reason for invalidity(we can rename it for more readability) and 
- An error in `OverseerFuncRes` is an internal error

in short, we need both

Reply:
Again, `ValidationResult` stores the validation result, and the `reason for invalidity` is also a result.

- If a candidate is valid, we store some data in `ValidationResult.`
- So, in my opinion, we should also store the `reason for invalidity` in the `ValidationResult` only. (also, it is more readable)

Still, if you want, we can remove `reason for invalidity` from `ValidationResult.`

Reply:
I've updated the code for `preformBasicChecks`  and renamed the error in `ValidationResult`.  Let me know your thought everyone.